Src/FluentAssertions/TypeEnumerableExtensions.cs

```diff
@@ -77,5 +77,60 @@ public static IEnumerable<Type> ThatImplement<T>(this IEnumerable<Type> types)
         {
             return new TypeSelector(types).ThatImplement<T>();
         }
+
+        /// <summary>
+        /// Filters to only include types that are classes.
+        /// </summary>
+        public static IEnumerable<Type> ThatAreClasses(this IEnumerable<Type> types)
+        {
+            return new TypeSelector(types).ThatAreClasses();
+        }
+
+        /// <summary>
+        /// Filters to only include types that are not classes.
+        /// </summary>
+        public static IEnumerable<Type> ThatAreNotClasses(this IEnumerable<Type> types)
+        {
+            return new TypeSelector(types).ThatAreNotClasses();
+        }
+        /// <summary>
+        /// Filters to only include types that are static classes.
+        /// </summary>
+        public static IEnumerable<Type> ThatAreStatic(this IEnumerable<Type> types)
+        {
+            return new TypeSelector(types).ThatAreStatic();
+        }
+
+        /// <summary>
+        /// Filters to only include types that are not static classes.
+        /// </summary>
+        public static IEnumerable<Type> ThatAreNotStatic(this IEnumerable<Type> types)
+        {
+            return new TypeSelector(types).ThatAreNotStatic();
+        }
+
+        /// <summary>
+        /// Filters to only include types that satisfies the <paramref name="predicate"/> passed.
+        /// </summary>
+        public static IEnumerable<Type> ThatAre(this IEnumerable<Type> types, Func<Type, bool> predicate)
```

Comment:
Thinking aloud.
Would it be better to name this `ThatSatisfies`?
@dennisdoomen 

```c#
types.ThatAre(t => t.IsCSharpStatic())
types.ThatSatisfies(t => t.IsCSharpStatic())
```

Reply:
In my opinion `ThatAre((t => t.IsCSharpStatic())` looks more fluid. But this is up to you.

Reply:
IMHO, I think `ThatSatisfy` is even better.

Reply:
You're right. This is the best option. Done.

Reply:
phew, thanks for catching my plural typo.
This also aligns better with the `SatisfyRespectively` name.